modules/Bio/EnsEMBL/Production/Pipeline/Ga4ghChecksum/CheckSumGenerator.pm

```diff
@@ -0,0 +1,206 @@
+
+=back
+
+The script is responsible for generating secure hash for genomic sequences 
+
+Allowed parameters are:
+
+=over 8
+
+=item hash_type - hash algorithm type 
+=item sequence_type - type of genomic sequences
+
+
+=back
+
+=cut
+
+package Bio::EnsEMBL::Production::Pipeline::Ga4ghChecksum::CheckSumGenerator;
+
+use strict;
+use warnings;
+use Digest::SHA;
+use Digest::MD5;
+use MIME::Base64 qw/encode_base64url/;
+use Encode;
+
+use Bio::EnsEMBL::Attribute;
+use Bio::EnsEMBL::Registry;
+use base qw/Bio::EnsEMBL::Production::Pipeline::Common::Base/;
+
+
+sub param_defaults {
+  my ($self) = @_;
+  return {
+   digest_size => 24,
+   chunk_size  => 100000,
+  };
+}
+
+sub fetch_input {
+  my ($self) = @_;
+  my $hash_types = $self->param('hash_types');
+  $hash_types = ['sha512t24u', 'md5'] if scalar(@$hash_types) == 0 ;
+  $self->param('hash_types', $hash_types);
+  return;
+}
+
+sub run {
+
+  my ($self) = @_;
+  my $hash_types = $self->param('hash_types');
+  my $type = $self->param('group');
+  my $species = $self->param('species');
+  my $seq_type = $self->param('seq_type');
+  my ($genome_ad, $genome_feature,  $sequence, $attrib_table, $genome_feature_id, $genome_slice) ;
+  my $dba  = $self->get_DBAdaptor($type);
+  my $registry = 'Bio::EnsEMBL::Registry';
+  if(! $dba) {
+    $self->info("Cannot find adaptor for type %s", $type);
+    next;
+  }
+  my $attribute_adaptor = $dba->get_adaptor('Attribute');
+  if( $seq_type eq 'toplevel'){
+    $genome_ad = $registry->get_adaptor( $species, $type, "Slice" );
+    $attrib_table = "seq_region";
+    $genome_feature_id = $self->param('seq_region_id');
+    $genome_slice = $genome_ad->fetch_by_seq_region_id($self->param('seq_region_id'));
+  }
+  elsif($seq_type eq 'cdna' || $seq_type eq 'cds' || $seq_type eq 'pep'){
+    $genome_ad = $registry->get_adaptor($species, $type, "Transcript");
+    $genome_feature = $genome_ad->fetch_by_dbID($self->param('transcript_dbid'));
+    $genome_slice = $genome_feature->seq; #create a seq object 
+    $attrib_table = "transcript";
+    $genome_feature_id = $self->param('transcript_dbid');
+
+    if( $seq_type eq 'cds' ){
+      $genome_slice->seq($genome_feature->translateable_seq);
+    }
+    if( $seq_type eq 'pep' ){
+      $genome_slice = $genome_feature->translate;
+      $attrib_table = "translation";
+      $genome_feature_id = $self->param('translation_dbid');
+    }	    
+  }else{
+    die "Unknown $seq_type : valid seq_types are toplevel, cdna, cds, pep";	  
+  }
+
+  #generate hash and update the attrb table 
+  foreach my $hash_method (@$hash_types){
+    my $at_code = $hash_method."_".$seq_type;
+    my $sequence_hash = $self->generate_sequence_hash($genome_slice, $hash_method);
+    $self->update_attrib_table($attribute_adaptor, $at_code, $sequence_hash, $genome_feature_id , $attrib_table);
+  }  
+}
+
+sub generate_sequence_hash {
+  my ($self, $slice, $hash_method) = @_;
+  my $sequence_hash;
+  if( $hash_method =~ /sha\-?(\d+)/){
```

Comment:
This seems unnecessary to me. If we have another algorithm e.g. `sha512t30u`, then wouldn't we specifically call that out rather than parsing out something which will a the moment always be 24?